service/src/main/java/com/commercetools/pspadapter/payone/mapping/MappingUtil.java

```diff
@@ -129,22 +129,30 @@ public static void mapCustomerToRequest(@Nonnull final AuthorizationRequest requ
 
     public static void mapShippingAddressToRequest(final AuthorizationRequest request, final Address shippingAddress) {
 
-        if(shippingAddress == null) {
-            throw new IllegalArgumentException("Missing shipping address details");
-        }
-
-        request.setShipping_firstname(shippingAddress.getFirstName());
-        request.setShipping_lastname(shippingAddress.getLastName());
-        request.setShipping_street(joinStringsIgnoringNull(Arrays.asList(shippingAddress.getStreetName(),
-            shippingAddress.getStreetNumber())));
-        request.setShipping_zip(shippingAddress.getPostalCode());
-        request.setShipping_city(shippingAddress.getCity());
-        request.setShipping_country(shippingAddress.getCountry().toLocale().getCountry());
-        request.setShipping_company(joinStringsIgnoringNull(Arrays.asList(shippingAddress.getCompany(),
-            shippingAddress.getDepartment())));
-
-        if (countriesWithStateAllowed.contains(shippingAddress.getCountry())) {
-            request.setShipping_state(shippingAddress.getState());
+        final String shippingCountry = shippingAddress.getCountry().toLocale().getCountry();
+        final String shipping_street = joinStringsIgnoringNull(asList(shippingAddress.getStreetName(),
+            shippingAddress.getStreetNumber()));
+
+        if(shippingAddress == null
+            || StringUtils.isBlank(shippingAddress.getPostalCode())
+            || StringUtils.isBlank(shippingCountry)
+            || StringUtils.isBlank(shipping_street)
+            || StringUtils.isBlank(shippingAddress.getCity())
+        ) {
+            request.setNoShipping(1);
```

Comment:
I wonder if it is correct to set it for all payment methods in general? Should not this be set for PayPal payments only?

Reply:
https://github.com/commercetools/commercetools-payone-integration/pull/413/commits/6f1625eeeb82416971e4847793c27599ec452c7e